### ISSUE TITLE
Borders on views with border radius set render incorrectly on Android Jellybean

### ISSUE THREAD

**Issue by jasonthomasgray** [facebook/react-native#18000]
### Is this a bug report?

Yes

### Have you read the [Contributing Guidelines](https://facebook.github.io/react-native/docs/contributing.html)?

Yes

### Environment

Environment:
  OS: Linux 4.13
  Node: 6.12.3
  Yarn: 1.3.2
  npm: 3.10.10
  Watchman: Not Found
  Xcode: N/A
  Android Studio: 3.0 AI-171.4443003

Packages: (wanted => installed)
  react: ^16.3.0-alpha.1 => 16.3.0-alpha.1
  react-native: ^0.54.0-rc.3 => 0.54.0-rc.3

Target Platform: Android Jellybean (API 16)

### Steps to Reproduce

https://github.com/jasonthomasgray/react-native-jellybean-issue
1. react-native init jellybeantest
2. add border styles to instructions in App.js
    ```js
    borderColor: '#222',
    borderRadius: 5,
    borderWidth: 1,
    ```
3. react-native run-android (on an emulator running API 16)

### Expected Behavior

The text should be surrounded with a border with rounded corners
![image](https://user-images.githubusercontent.com/17647414/36304943-79ebe6a8-1309-11e8-9950-9185c3814330.png)
API 19 | API 23

### Actual Behavior

The border is variable width and has no rounded corners.
![image](https://user-images.githubusercontent.com/17647414/36305184-6a4ce69c-130a-11e8-97f8-0819f3627e83.png)

### Reproducible Demo

https://github.com/jasonthomasgray/react-native-jellybean-issue
Launch the app on an emulator running android API 16 to see the issue

**Comment by AkilLis** [comment]
Same issue as well. Tested /0.52.0, 0.53.2, 0.54/

**Comment by pcelestino** [comment]
This bug is not present in version 0.50.4

**Comment by AkilLis** [comment]
So point is this issue not gonna fixed next versions as well?. Pathetic ...

**Comment by hramos** [comment]
@AkilLis your latest comment on this issue is disrespectful and not constructive. Please take a look at the [Code of Conduct](https://github.com/facebook/react-native/blob/master/CODE_OF_CONDUCT.md) before engaging in this repository. This is your first warning.

**Comment by himabindugadupudi** [comment]
Fixed with commit c5ca26a which is now in Master. Please test with Master build.
